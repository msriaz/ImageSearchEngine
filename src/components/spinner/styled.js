/**
 * Spinner Variants for different sizes
 */
export const SPINNER_VARIANTS = {
  SMALL: 'small',
  LARGE: 'large',
};

/**
 * Spinner Sizes
 */
export const SPINNER_SIZE = {
  [SPINNER_VARIANTS.SMALL]: 'small',
  [SPINNER_VARIANTS.LARGE]: 'large',
};
