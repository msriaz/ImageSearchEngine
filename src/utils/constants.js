import {color as colors} from '../theme';
import {generateColorPalleteConstants} from './index';

export const APP_COLORS = generateColorPalleteConstants(colors);

export const STORAGE_KEYS = {
  HISTORY: 'history',
};
