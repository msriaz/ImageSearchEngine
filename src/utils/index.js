import {useEffect, useRef} from 'react';
import snakeCase from 'lodash/snakeCase';
/**
 *
 * @param {*} value
 * @returns previous state
 */
export const usePrevious = value => {
  const ref = useRef();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
};

/**
 *
 * @param {*} imgData
 * @returns image url to render
 */
export const getImageUrl = imgData => {
  const {farm, server, id, secret} = imgData;
  return `https://farm${farm}.static.flickr.com/${server}/${id}_${secret}.jpg`;
};

/**
 * generate color consts from color pallette
 * @param {*} colors
 */
export const generateColorPalleteConstants = colors => {
  const colorsList = {};
  Object.keys(colors).forEach(colorKey => {
    colorsList[snakeCase(colorKey).toUpperCase()] = colors[colorKey];
  });
  return colorsList;
};

/**
 *
 * @param {*} funcName
 * @param {*} error
 */
export const reportError = (funcName, error) => {
  console.log(`Error in ${funcName}`, error);
};
