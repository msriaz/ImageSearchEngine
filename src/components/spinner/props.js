import PropTypes from 'prop-types';
import {SPINNER_VARIANTS} from './styled';

export const propTypes = {
  /**
   * Color from themes, is an enum
   */
  color: PropTypes.string,
  /**
   * Size of the spinner, "small" or "large"
   */
  variant: PropTypes.oneOf(Object.values(SPINNER_VARIANTS)),
};

export const defaultProps = {
  variant: SPINNER_VARIANTS.SMALL,
};
