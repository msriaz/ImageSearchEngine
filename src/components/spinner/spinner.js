import React from 'react';
import {ActivityIndicator} from 'react-native';
import {defaultProps, propTypes} from './props';

import {SPINNER_VARIANTS, SPINNER_SIZE} from './styled';

/**
 * Spinner Component
 * Simple wrapper around native ActivityIndicator
 * @param {color} defines color of the spinner
 * @param {variant} defines size of the spinner ["large", "small"]
 * @param {...props} defines additional props to be passed to ActivityIndicator
 */

export const Spinner = ({color, variant, ...props}) => {
  const size = SPINNER_SIZE[variant];

  return <ActivityIndicator color={color} size={size} {...props} />;
};

Spinner.propTypes = propTypes;
Spinner.defaultProps = defaultProps;

Spinner.VARIANTS = SPINNER_VARIANTS;
