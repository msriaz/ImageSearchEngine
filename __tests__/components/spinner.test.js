import React from 'react';

import {render} from '../utils/theme-wrapper';
import {Spinner} from '../../src/components/spinner';

describe('Spinner Component', () => {
  it('should match the Spinner snapshot', () => {
    const compTree = render(<Spinner />);
    expect(compTree.toJSON()).toMatchSnapshot();
  });
});
