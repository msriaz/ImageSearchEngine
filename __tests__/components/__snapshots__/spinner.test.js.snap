// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should match the Spinner snapshot 1`] = `
<ActivityIndicator
  size="small"
/>
`;
